### PR TITLE
linting clean up

### PR DESCRIPTION
**Pull request by cornfeedhobo**
## Description

- Lints core files
  - bash_it.sh
  - lib/appearance.bash
- Ensures all tests/assertions are done using double bracket expansions (`[[]]`)
- Minimal fixes needed to comply with `shellcheck`

Commits are broken up for comprehension since looking at the final diff will be difficult to get through.

## Motivation and Context

I want to make some changes to the core, but the lack of existing linting is making that difficult.

## How Has This Been Tested?

- Locally
- `bash-it doctor`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] If I have added a new file, I also added it to ``clean_files.txt`` and formatted it using ``lint_clean_files.sh``.
- [ ] I have added tests to cover my changes, and all the new and existing tests pass.

### CODE REVIEW

**Comment by cornfeedhobo** [comment]
Since this formats `history.bash`, please don't merge until #1764 is resolved.

**Comment by nwinkler** [comment]
This is great! Looking forward to getting the majority of the codebase in compliance with our guidelines!

File: bash_it.sh [hunk 1]
```diff
@@ -55,100 +53,95 @@ done
...
-if ! command -v reload &>/dev/null && [ -n "$BASH_IT_RELOAD_LEGACY" ]; then
-  case $OSTYPE in
-    darwin*)
-      alias reload='source ~/.bash_profile'
-      ;;
-    *)
-      alias reload='source ~/.bashrc'
-      ;;
-  esac
+if ! command -v reload &> /dev/null && [[ -n "$BASH_IT_RELOAD_LEGACY" ]]; then
```
Comment (NoahGorny) [inline bash_it.sh:136]: we can use `command_exists` here

File: lib/history.bash [hunk 2]
```diff
@@ -7,6 +7,6 @@ export HISTCONTROL=${HISTCONTROL:-ignorespace:erasedups} # erase duplicates; alt
 export HISTSIZE=${HISTSIZE:-5000}                        # resize history size
 export AUTOFEATURE=${AUTOFEATURE:-true autotest}         # Cucumber / Autotest integration
 
-function rh {
-  history | awk '{a[$2]++}END{for(i in a){print a[i] " " i}}' | sort -rn | head
+rh() {
```
Comment (NoahGorny) [inline lib/history.bash:10]: I guess this needs a rebase

**Review by NoahGorny — CHANGES_REQUESTED** [review]
Left some small comments, otherwise LGTM

**Pull request closed without merging**